httpcore/src/main/java/org/apache/http/impl/pool/BasicConnFactory.java

```diff
@@ -177,7 +181,24 @@ public HttpClientConnection create(final HttpHost host) throws IOException {
             socket.setSoLinger(true, linger);
         }
         socket.setKeepAlive(this.sconfig.isSoKeepAlive());
-        socket.connect(new InetSocketAddress(hostname, port), this.connectTimeout);
+        // Run this under a doPrivileged to support lib users that run under a SecurityManager this allows granting connect permissions
+        // only to this library
+        final Socket finalSocket = socket;
```

Comment:
@s1monw Can we eliminate `finalSocket`? It is no longer required.

Reply:
done